### PR TITLE
add zipApply, fixes #8102

### LINKED ISSUE #8102: Module `sequtils`: should provide `mzip`
As in subject, example:
```
import sequtils
var array1 = [1, 2, 3, 4]
var array2 = [3, 4, 5, 6]
for x in mzip(array1, array2):
  x.b += x.a
```

### PR DESCRIPTION
**Pull request by hlaaftana**
Fixes #8102

### CODE REVIEW

File: lib/pure/collections/sequtils.nim [hunk 1]
```diff
@@ -277,6 +277,61 @@ proc apply*[T](s: var openArray[T], op: proc (x: T): T {.closure.})
...
+  ##
+  ## Note that this requires your input and output types to
+  ## be the same, since they are modified in-place.
+  ## If one container is shorter, the remaining items in the
+  ## longer container are discarded.
+  ## The parameter function takes both a ``var S`` and a ``var T`` type parameter.
+  ##
+  ## Example:
+  ##
+  ## .. code-block:: nim
```
Comment (dom96) [inline lib/pure/collections/sequtils.nim]: These should be runnable examples

Comment (hlaaftana) [reply lib/pure/collections/sequtils.nim]: Fixed

**Comment by Varriount** [comment]
What's the benefit of using this over zip and apply separately?

**Comment by krux02** [comment]
This does not fix anything.

**Comment by krux02** [comment]
I am sorry. You did provide a solution to the original problem. But I would argue that it is a fix. The problem is that mZip can't be implemented. I just wrote in the original issue the error message that I got. You offer a workaround so that you don't need an mZip iterator. So you should post this workaround in the issue as a temporary solution. But I am agains it it be part of the standard library.

**Comment by Araq** [comment]
So ``zipApply`` has a type signature that spans 2 lines. 6 lines of documentation. 8 lines of examples. And the implementation is 2 lines (and can be simplified to one). Sorry, -1 from me. That's exactly the bloat that I would like to avoid in the standard library.

**Pull request closed without merging**